client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts

```diff
@@ -533,13 +551,17 @@ function createItemToAddToCart( {
 		createRequestCartProduct( {
 			product_id: productId,
 			product_slug: productSlug,
-		} )
+		} ),
+		{ isJetpackUserlessCheckout }
 	);
 }
 
-function addContextToProduct( product: RequestCartProduct ): RequestCartProduct {
+function addContextToProduct(
+	product: RequestCartProduct,
+	options?: Record< string, unknown >
```

Comment:
I think these aren't unknown, right? They have to be something in `RequestCartProductExtra`. Maybe:

```suggestion
	options?: Partial< RequestCartProductExtra >
```